src/submissionService.ts

```diff
@@ -41,3 +41,47 @@ export function getRootElementValueById(
     }
   }
 }
+
+/**
+ * Takes an element name and a submission object, and returns the provided
+ * element's value as a string. Used for replaceable values in OneBlink
+ * Calculation and Info (HTML) elements.
+ *
+ * #### Example
+ *
+ * ```js
+ * const nameElementValue = submissionService.getSubmissionValueAsString(
+ *   'Name_Element',
+ *   submission,
+ * )
+ * ```
+ *
+ * @param elementName
+ * @param submission
+ * @returns
+ */
+export function getSubmissionValueAsString(
+  elementName: string,
+  submission: SubmissionTypes.S3SubmissionData['submission'],
+): string {
+  const v = submission[elementName]
```

Comment:
During a submission the value could be on the model but not shown, don't know if it's here we want to take care of that on in apps-react.  It's not an issue post submission as the model will have been scrubbed.

Reply:
Mm,yeh I think we handle it in apps-react, and pass in the clean submission model

Reply:
I would just use the parent form elements conditionally shown object, much faster.